src/main/java/org/mockito/verification/VerificationWrapperInOrderWrapper.java

```diff
@@ -0,0 +1,49 @@
+package org.mockito.verification;
+
+import org.mockito.exceptions.base.MockitoException;
+import org.mockito.internal.InOrderImpl;
+import org.mockito.internal.verification.InOrderWrapper;
+import org.mockito.internal.verification.VerificationOverTimeImpl;
+import org.mockito.internal.verification.api.VerificationData;
+import org.mockito.internal.verification.api.VerificationInOrderMode;
+
+public class VerificationWrapperInOrderWrapper implements VerificationMode {
+    private final VerificationMode delegate;
+
+    public VerificationWrapperInOrderWrapper(VerificationWrapper<?> verificationWrapper, InOrderImpl inOrder) {
+        VerificationMode verificationMode = verificationWrapper.wrappedVerification;
+
+        VerificationMode inOrderWrappedVerificationMode = wrapInOrder(verificationWrapper, verificationMode, inOrder);
+
+        delegate = verificationWrapper.copySelfWithNewVerificationMode(inOrderWrappedVerificationMode);
+    }
+
+    @Override
+    public void verify(VerificationData data) {
+        delegate.verify(data);
+    }
+
+    @Override
+    public VerificationMode description(String description) {
+        return null;
+    }
+
+    private VerificationMode wrapInOrder(VerificationWrapper<?> verificationWrapper, VerificationMode verificationMode, InOrderImpl inOrder) {
+        if (verificationMode instanceof VerificationInOrderMode) {
+            final VerificationInOrderMode verificationInOrderMode = (VerificationInOrderMode)verificationMode;
+            return new InOrderWrapper(verificationInOrderMode, inOrder);
+        } else if (verificationMode instanceof VerificationOverTimeImpl) {
+            final VerificationOverTimeImpl verificationOverTime = (VerificationOverTimeImpl)verificationMode;
+            if (verificationOverTime.isReturnOnSuccess()) {
+                return new VerificationOverTimeImpl(verificationOverTime.getPollingPeriodMillis(),
```

Comment:
@Jazzepi The readability should be improved here.You should pass local variables as parameter to the constructor. You could also consider to move the whole construction process into a new method because of its complexity e.g. `wrapVerificationOverTime(..)` .